addons/mrp/models/mrp_production.py

```diff
@@ -1963,7 +1974,7 @@ def button_mark_done(self):
             })
         return action
 
-    def _pre_button_mark_done(self):
+    def pre_button_mark_done(self):
```

Comment:
Why is it now a public method?

Reply:
Because, we call it from the client directly in OE. In the MES the MO/WO validation doesn't apply directly but after a 10s fadeout, we don't want to raise the error message 10s later and do the check directly. So we split the validation check and the validation itself